src/ir/context.rs

```diff
@@ -569,6 +569,24 @@ impl<'ctx> BindgenContext<'ctx> {
         self.used_template_parameters = Some(used_params);
     }
 
+    /// Return `true` if `item` uses the given `template_param`, `false`
+    /// otherwise.
+    ///
+    /// This method may only be called during the codegen phase, because the
+    /// template usage information is only computed as we enter the codegen
+    /// phase.
+    pub fn uses_template_parameter(&self, item: ItemId, template_param: ItemId) -> bool {
+        assert!(self.in_codegen_phase(),
```

Comment:
I think even better would be to have different `Context` types and get rid of that nasty transmute. Then we could simply have methods that require more info on the context that has that info by construction, and push all these asserts into the type system at compile time.